prow/plank/controller_test.go

```diff
@@ -353,12 +354,42 @@ func TestSyncNonPendingJobs(t *testing.T) {
 					State: kube.TriggeredState,
 				},
 			},
-			err:              kube.UnprocessableEntityError(errors.New("no way jose")),
+			podErr:           kube.UnprocessableEntityError{errors.New("no way jose")},
 			expectedState:    kube.ErrorState,
 			expectedComplete: true,
 			expectedReport:   true,
 		},
 		{
+			name: "conflict error starting pod",
+			pj: kube.ProwJob{
+				Spec: kube.ProwJobSpec{
+					Job:  "boop",
+					Type: kube.PeriodicJob,
+				},
+				Status: kube.ProwJobStatus{
+					State: kube.TriggeredState,
+				},
+			},
+			podErr:        kube.ConflictError{errors.New("no way jose")},
```

Comment:
The most common issues for being unable to create a pod are lack of quota or missing rbac/podsecuritypolicy permissions. That being said, I am fine with leaving this as is until we add a new error type that is more relevant to pod creation failures.